services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/dao/NmmInfoRepository.java

```diff
@@ -0,0 +1,186 @@
+    private ICache<String, NmmInfo> cache;
+
+    @Autowired
+    public NmmInfoRepository(CacheFactory cacheFactory) {
+        cache = cacheFactory.getCache(NmmInfo.class);
+    }
+
+    public ICache<String, NmmInfo> getCache() {
+        return cache;
+    }
+
+    @PostConstruct
+    private void init() {
+        logger.info("NmmInfoRepository init completed");
+    }
+
+    /**
+     * find an NmmInfo item from the repository.
+     *
+     * @param id NcmId
+     * @return NmmInfo
+     * @throws CacheException exception or DbException
+     */
+    public NmmInfo findItem(String id) throws CacheException {
+        return cache.get(id);
+    }
+
+    /**
+     * find information about all NCM
+     *
+     * @return NmmInfo
+     * @throws CacheException or DbException
+     */
+    public Map findAllItems() throws CacheException {
+        return cache.getAll();
+    }
+
+    /**
+     * find all NcmInfo information by filter: Not needed and not supported.
+     * @param queryParams url request params
+     * @return Map of NmmInfo items
+     * @throws CacheException exception
+     */
+    public Map<String, NmmInfo> findAllItems(Map<String, Object[]> queryParams) throws CacheException {
+        String error = "NCM query by filter is invalid operation";
+        logger.error(error);
+        throw new CacheException(error);
+    }
+
+    /**
+     * add a new NMM info to repository
+     *
+     * @param nmmInfo new NMM information
+     * @throws CacheException or DbException
+     */
+    public void addItem(NmmInfo nmmInfo) throws CacheException {
+        String ncmId = nmmInfo.getNcmId();
+        logger.info("Add an NmmInfo entry, NCM Id:" + ncmId);
+
+        // TODO: Push Transaction out of DAO.
+        // Transaction should cover DPM, NCM and local store.
+        // This means NMM will have to maintain "Transaction context" covering DPM, NCM and local
+        // changes.
+        // Having to initiate a transaction in DPM, NCM and local store prior to modification
+        // and following it by a commit or rollback is very much like Two Phase Commit: tricky
+        // and expensive.
+        // There is also the problem of "obtaining" a "transaction context" from DPM and NCM which
+        // are physically separate processes possibly running on different machines. This is a roadmap
+        // item.
+        try (Transaction tx = cache.getTransaction().start()) {
+            cache.put(ncmId, nmmInfo);
+            tx.commit();
+        } catch (CacheException e) {
+            throw e;
+        } catch (Exception e) {
+            logger.error("Add an NmmInfo entry error: "+e.getMessage());
+        }
+    }
+
+    /**
+     * add a new NMM info to repository
+     *
+     * @param ncmId, NCM to which new nodes are being appended
+     * @param nodeIds, the new nodes
+     * @throws CacheException or DbException
+     */
+    public void appendNodes(String ncmId,  List<String> nodeIds) throws CacheException {
```

Comment:
@pkommoju Since you have `appendNodes` here, do we need to support `removeNodes` API?

Reply:
Of course. I will add it.

Reply:
Added it. Removed two unsupported methods.
To add unit tests there are no service end points in this PR. Next PR will add the service API and the unit tests.